src/core/lib/surface/completion_queue.c

```diff
@@ -449,7 +449,7 @@ grpc_completion_queue *grpc_completion_queue_create_internal(
 static void cq_init_next(void *ptr) {
   cq_next_data *cqd = (cq_next_data *)ptr;
   /* Initial count is dropped by grpc_completion_queue_shutdown */
-  gpr_atm_no_barrier_store(&cqd->pending_events, 1);
+  gpr_atm_rel_store(&cqd->pending_events, 1);
```

Comment:
Is there really a reason to have release semantic in the CQ init function? By definition, some other thread can't see this item until there's been some intervening synchronization operation

Reply:
No reason. No barrier is just fine. Actually there is no reason to even do an atomic operation here - since we know for sure that only one thread will ever call this.